include/xgboost/data.h

```diff
@@ -169,8 +169,14 @@ class SparsePage {
   inline Inst operator[](size_t i) const {
     const auto& data_vec = data.HostVector();
     const auto& offset_vec = offset.HostVector();
+    size_t size;
```

Comment:
Could you leave a comment here, maybe a link to this issue?  Otherwise silencing an index overbound error feels weird to me .

Reply:
Actually, can we limit it to distributed mode only?

Reply:
addressed